File: examples/python/message_reports.py
import os
import sys

from kalourmade_arkesel import ArkeselClient

ids = sys.argv[1:] or sys.exit("Usage: python message_reports.py <id> [id...]")
client = ArkeselClient(os.environ["ARKESEL_API_KEY"])

reports = client.sms.message_reports(ids)

for id, report in reports.items():
    if report.error is not None:
        print(f"{id} — error: {report.error}")
    else:
        print(f"{id} — {report.message} ({report.status.value})")

File: examples/node/messageReports.ts
import { ArkeselClient } from '@kalourmade/arkesel-sms';

const ids = process.argv.slice(2);
if (ids.length === 0) {
  console.error('Usage: tsx messageReports.ts <id> [id...]');
  process.exit(1);
}

const client = new ArkeselClient(process.env.ARKESEL_API_KEY!);
const reports = await client.sms.messageReports(ids);

for (const [id, report] of Object.entries(reports)) {
  if (report.error !== null) {
    console.log(`${id} — error: ${report.error}`);
  } else {
    console.log(`${id} — ${report.message} (${report.status})`);
  }
}
